controllers/datadogagent/dependencies/store.go

```diff
@@ -205,6 +206,13 @@ func (ds *Store) Apply(ctx context.Context, k8sClient client.Client) []error {
 				continue
 			}
 
+			// Special case for Service. The cluster IPs should be immutable, so don't
```

Comment:
This is something that we were already doing for V1:
https://github.com/DataDog/datadog-operator/blob/24d3c0966e8be89a93fe6f6e529f71703b0e240a/controllers/datadogagent/service.go#L290

The integration tests fail without this.